suite-native/graph/src/components/TransactionEventTooltip.tsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import Animated, { FadeIn, FadeOut } from 'react-native-reanimated';
+import { Dimensions } from 'react-native';
+
+import { G, N } from '@mobily/ts-belt';
+
+import { Card, Text } from '@suite-native/atoms';
+import { CryptoAmountFormatter, SignValueFormatter } from '@suite-native/formatters';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { GroupedBalanceMovement } from '@suite-common/graph';
+
+export type TransactionEventTooltipProps = GroupedBalanceMovement['payload'] & {
+    eventX: number | null;
+    eventY: number | null;
+};
+
+const ANIMATION_DURATION = 200;
+const SCREEN_WIDTH = Dimensions.get('screen').width;
+
+const TooltipContainerStyle = prepareNativeStyle<{ x: number; y: number }>((_, { x, y }) => ({
+    position: 'absolute',
+    left: x + 25,
+    top: N.clamp(0, 100, y), // The clamping prevents an Y axis overflow.
+    zIndex: 10000,
+    extend: {
+        // If the tooltip is on the right half of the screen,
+        // render it on the left side of the referred event point to prevent an X axis overflow.
+        condition: x > SCREEN_WIDTH / 2,
+        style: {
+            left: x - 145,
+        },
+    },
+}));
+
+const TooltipCardStyle = prepareNativeStyle(() => ({ paddingVertical: 12 }));
+
+export const TransactionEventTooltip = ({
+    eventX,
+    eventY,
+    networkSymbol,
+    received,
+    sent,
+    receivedTransactionsCount,
+    sentTransactionsCount,
+}: TransactionEventTooltipProps) => {
+    const { applyStyle } = useNativeStyles();
+    if (G.isNull(eventX) || G.isNull(eventY)) {
+        return null;
+    }
+
+    const totalAmount = received && sent ? received - sent : null;
+
+    const isSentDisplayed = sent !== 0 && sentTransactionsCount !== 0;
+    const isReceivedDisplayed = received !== 0 && receivedTransactionsCount !== 0;
+
+    return (
+        <Animated.View
+            style={applyStyle(TooltipContainerStyle, { x: eventX, y: eventY })}
+            entering={FadeIn.duration(ANIMATION_DURATION)}
+            exiting={FadeOut.duration(ANIMATION_DURATION)}
+        >
+            <Card style={applyStyle(TooltipCardStyle)}>
+                {isSentDisplayed && (
+                    <>
+                        <Text variant="label" color="textSubdued">
+                            Sent · {sentTransactionsCount}
+                        </Text>
+                        <Text>
+                            <SignValueFormatter value="negative" variant="label" />
+                            <CryptoAmountFormatter
+                                color="textDefault"
+                                variant="label"
+                                value={sent}
+                                network={networkSymbol}
+                                isBalance={false}
+                            />
+                        </Text>
```

Comment:
Make this a component and reuse in all 3 places

Reply:
90955f4fc